lib/segment/src/index/field_index/histogram.rs

```diff
@@ -267,7 +337,13 @@ impl<T: Numericable> Histogram<T> {
                     } else {
                         0
                     };
-                    let max_estimate = a_count.right + 1;
+                    let mut max_estimate = a_count.right + 1;
+
+                    if b_count.right == 0 {
+                        // This is the last border, so we need to include it
+                        max_estimate += 1;
+                    }
```

Comment:
This change is not related to this PR now that we are using the btree stream approach, but I'm pretty sure it is a correct fix... _maybe_?